script/hassfest/manifest.py

```diff
@@ -343,6 +337,15 @@ def validate_manifest(integration: Integration, core_components_dir: Path) -> No
             "Virtual integration points to non-existing supported_by integration",
         )
 
+    if (quality_scale := integration.manifest.get("quality_scale")) in {
+        "gold",
```

Comment:
Side note idea: we could add an IntEnum representing the quality scale to be able to compare version higher or lower than a quality scale (in this case higher than silver).